dataprofiler/data_readers/csv_data.py

```diff
@@ -581,15 +576,16 @@ def is_match(cls, file_path, options=None):
         if options is None:
             options = dict()
 
-        file_encoding = None
-        if not isinstance(file_path, StringIO):
-            file_encoding = data_utils.detect_file_encoding(file_path=file_path)
+        if 'encoding' not in options and not isinstance(file_path, StringIO):
+            options['encoding'] = data_utils.detect_file_encoding(
```

Comment:
Is it appropriate to be modifying options that the user passes us rather than simply executing logic in accordance with them? I don't know if this occurs elsewhere but it feels strange to change the options as they are an indication of user intention. I am open to discussion though.

Reply:
it is a technique we use so that when we pass information to the Profiler, the options don't have to recompute the encoding. Also, note that it doesn't overwrite the options.

Reply:
In future, we could do something like:
```python
is_match = BlankData.is_match(..., options)
# or
is_match, options = BlankData.is_match(..., options, update_options=True)
```